client-app/shared/cart/components/add-to-cart.vue

```diff
@@ -106,10 +106,12 @@ async function onChange() {
   let updatedCart: ShortCartFragment | undefined;
 
   const isAlreadyExistsInTheCart = !!lineItem;
-  const mode = isAlreadyExistsInTheCart && !isConfigurable.value ? AddToCartModeType.Update : AddToCartModeType.Add;
+  const mode = isAlreadyExistsInTheCart ? AddToCartModeType.Update : AddToCartModeType.Add;
 
   if (mode === AddToCartModeType.Update) {
-    updatedCart = await changeItemQuantity(lineItem!.id, enteredQuantity.value || 0);
+    updatedCart = isConfigurable.value
+      ? await changeCartConfiguredItem(lineItem!.id, enteredQuantity.value || 0, selectedConfigurationInput.value)
```

Comment:
! try not to use this

Reply:
![image](https://github.com/user-attachments/assets/b52e081f-4151-493a-ae35-bf94db669cef)
there is such a schema, which implicitly checked if lineItem exists, but we need to use ! as it too complicated for TypeScript to realize that :)